### PR TITLE
docs: added new macros for pack PEM-1603

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
## Describe the Change

This PR adds new macros related to the cluster profile. The macros have also been updated with descriptions.

## Review Changes

💻 [Add Preview URL](https://deploy-preview-1250--docs-spectrocloud.netlify.app/registries-and-packs/pack-constraints#packmacros)

🎫 [PEM-1606](https://spectrocloud.atlassian.net/browse/PEM-1606)

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1250#" title="Author self-approved">karl-cardenas-coding</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by ritawatson — APPROVED** [review]
LGTM.

**Pull request merged**
